File: Spring/szakdogaservice/szakdogaservice/src/main/java/hu/szurdok/szakdogaservice/group/Group.java
package hu.szurdok.szakdogaservice.group;

import lombok.AllArgsConstructor;
import lombok.Data;

import javax.persistence.*;

@Entity
@Data
@AllArgsConstructor
@Table(name = "groups")
public class Group {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Basic
    @Column(name = "name")
    private String name;

    @Basic
    @Column(name = "owner_id")
    private String ownerId;

    @Basic
    @Column(name = "join_code")
    private String joinCode;

    @Basic
    @Column(name = "picture")
    private String picture;
}

File: Android/Todo/app/src/main/java/hu/szurdok/todoapp/data/group/Group.kt
package hu.szurdok.todoapp.data.group

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Group (
    @PrimaryKey
    val id : Int,
    val name : String,
    val joinCode : String,
    val picture : String,
    val ownerId : String
) {
    override fun toString(): String {
        return "$id - $joinCode - $picture - $ownerId"
    }
}